src/client/opamCommands.ml

```diff
@@ -408,17 +416,31 @@ let init cli =
         OpamStd.Sys.guess_dot_profile shell >>| OpamFilename.of_string)
     in
     let cygwin_setup =
-      match cygwin_internal, cygwin_location with
-      | `internal, Some _ ->
+      match cygwin_internal, cygwin_location, cygwin_extra_packages with
+      | `internal , Some _, _ ->
         OpamConsole.error_and_exit `Bad_arguments
           "Options --cygwin-internal-install and \
            --cygwin-location are incompatible";
-      | `no, Some _ ->
-        OpamConsole.note "Ignoring argument --cygwin-location";
-        Some `no
-      | `none, None -> None
-      | (`default_location | `none), Some dir -> Some (`location dir)
-      | (`internal | `default_location | `no) as setup, None -> Some setup
+      | `default_location, _, Some _ ->
+        OpamConsole.error_and_exit `Bad_arguments
+          "Options --cygwin-local-install and \
+           --cygwin-extra-packages are incompatible";
+      | `no, Some _, _ ->
+        OpamConsole.error_and_exit `Bad_arguments
+          "Options --no-cygwin-setup and \
+           --cygwin-location are incompatible";
```

Comment:
In previous code, the behaviour was : _if `--no-cygwin-setup` is given, all other argument are ignored_. If there a reason to change that to bad arguments errors ?

Reply:
It was from https://github.com/ocaml/opam/pull/5930#discussion_r1579012410 (and for discussion!) - I think the previous behaviour is "bad"... if the user has specified two conflicting options, we should display an error.

Reply:
fair enough